subsys/bluetooth/host/conn.c

```diff
@@ -1222,10 +1223,13 @@ void bt_conn_set_state(struct bt_conn *conn, bt_conn_state_t state)
 			 * the application through bt_conn_disconnect or by
 			 * timeout set by bt_conn_le_create_param.timeout.
 			 */
-			if (conn->err) {
-				notify_connected(conn);
+			if (IS_ENABLED(CONFIG_BT_CENTRAL)) {
+				bt_le_scan_clear_reason_and_update(
+					BT_LE_SCAN_ENABLED_REASON_SCAN_BEFORE_INITIATE);
```

Comment:
If this really can't fit on a single line (and still be max 100 chars) it's an indication of this function either needing some refactoring, or using too long symbol/macro names :)

Reply:
Removed part of the MACRO names, that was not necessary.
Now I think this line is the only one that does not fit into 100 characters.

I don't really think that if this does not fit into 100 characters is necessary a sign of complexity. I would argue that descriptive function and enum field names are more important than having 8 whitespace indentation and 100 characters max lines.

I can further refine the naming of the function / the enum fields if you like. If so, would you have a suggestion on what to call it?

Reply:
> descriptive function and enum field names are more important

Whaaaat, you don't like `struct net_buf *buf`? how could you!

Maybe `bt_scan_add_user()` and `bt_scan_remove_user()` would work? If you're still looking to change the name :)

You can also remove the `SCAN_ENABLED_` from the enum names, as I suspect it's only there because our IDEs are not that good.

But this is really hiding the root cause which, agreeing with @jhedberg here, is complexity. 
This function is horrible to read (come at our desk we printed it out to prove a point lol). Chunking it into at least two functions would be a good start.

Reply:
> You can also remove the SCAN_ENABLED_ from the enum names, as I suspect it's only there because our IDEs are not that good.

I think that was a suggestion from me, and IIRC we do have a point in the BT guidelines issue about that :) 

> But this is really hiding the root cause which, agreeing with @jhedberg here, is complexity.
> This function is horrible to read (come at our desk we printed it out to prove a point lol). Chunking it into at least two functions would be a good start.

We really should just remove this :upside_down_face: